packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -14,6 +14,15 @@ function aliasIdToElementId(args, props) {
   }
 }
 
+// Use `_targetObject` to avoid stomping on a CP
+// that exists in the component
+function privatizeTargetObject(props) {
+  if (props.targetObject) {
+    props._targetObject = props.targetObject;
+    delete props.targetObject;
```

Comment:
Looks like it's an EmptyObject, so should be fine. But I've just gone ahead and set this to undefined.